### PR TITLE
[android] Implement android test runner for core unit tests

### PR DESCRIPTION
**Pull request by alexshalamov**
Private APIs from [android_runtime.so](https://android.googlesource.com/platform/bionic/+/master/android-changes-for-ndk-developers.md#private-api-enforced-for-api-level-24) are no longer accessible, thus, core unit tests are segfaulting on android platform.

This PR introduces simple unit test runner and VM initialization code that can be reused for benchmark tests.

### CODE REVIEW

**Comment by alexshalamov** [comment]
Most of the tests are passing, except:
- Annotations.AntimeridianAnnotationLarge (pixel diff)
- Expression.IsExpression (for some reason loads style outside of fixtures)
- Memory.* (have to figure out)
- Map.Offline (hangs when network status changes to online)

`make run-android-core-test-arm-v8--Map.Offline:Memory.*:Annotations.AntimeridianAnnotationLarge:Expression.IsExpression`

File: platform/android/src/test/runtime.cpp [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+namespace {
+const std::string kClassPathCommand{"--class_path="};
+const std::string kClassPath{"-Djava.class.path="};
+const std::string kDefaultDex{"/data/local/tmp/core-tests/classes.dex"};
+} // namespace
+
+namespace mbgl {
+namespace android {
+
+bool init_runtime(int argc, char *argv[]) {
```
Comment (kkaefer) [inline platform/android/src/test/runtime.cpp]: nit: we tend to use `camelCase` naming

Comment (alexshalamov) [reply platform/android/src/test/runtime.cpp]: :+1: done

**Comment by alexshalamov** [comment]
@kkaefer There are few failing cases due to small pixel mismatch differences in rendering, I couldn't figure out how to align all platforms to produce stable results. I've added d909e19 to filter-out some of the failures. Is this fine or maybe you would have better idea how to tackle it?

**Pull request merged**